examples/simple-genai-server/main.go

```diff
@@ -0,0 +1,240 @@
+	prompt := flag.String("Prompt", "", "The first prompt for the GenAI endpoint")
+	simEndpoint := flag.String("simEndpoint", "", "The full base URL to send API requests to to simulate user input")
+	simContext := flag.String("SimContext", "", "Context for the Sim endpoint")
+	numChats := flag.Int("NumChats", 1, "Number of back and forth chats between the sim and genAI")
+
+	var simConn *connection
+	var compConn * connection
+
+	flag.Parse()
+	if ep := os.Getenv("PORT"); ep != "" {
+		port = &ep
+	}
+	if sc := os.Getenv("SimContext"); sc != "" {
+		simContext = &sc
+	}
+	if gac := os.Getenv("GenAiContext"); gac != "" {
+		genAiContext = &gac
+	}
+	if p := os.Getenv("Prompt"); p != "" {
+		prompt = &p
+	}
+	if se := os.Getenv("SimEndpoint"); se != "" {
+		simEndpoint = &se
+		log.Printf("Creating Client at endpoint %s", *simEndpoint)
+		simConn = initClient(*simEndpoint, *simContext)
+	}
+	if gae := os.Getenv("GenAiEndpoint"); gae != "" {
+		genAiEndpoint = &gae
+		log.Printf("Creating Client at endpoint %s", *genAiEndpoint)
+		compConn = initClient(*genAiEndpoint, *genAiContext)
+	}
+	if nc := os.Getenv("NumChats"); nc != "" {
+		num, err := strconv.Atoi(nc)
+		if err != nil {
+			log.Fatalf("Could not parse NumChats: %v", err)
+		}
+		numChats = &num
+	}
+
+	log.Print("Creating SDK instance")
+	s, err := sdk.NewSDK()
+	if err != nil {
+		log.Fatalf("Could not connect to sdk: %v", err)
+	}
+
+	log.Print("Starting Health Ping")
+	// TODO: Is "cancel" still being used even if no method explicitly uses it?
+	ctx, _ := signals.NewSigKillContext()
+	go doHealth(s, ctx)
+
+	log.Print("Marking this server as ready")
+	if err := s.Ready(); err != nil {
+		log.Fatalf("Could not send ready message")
+	}
+
+	// Start up TCP listener so the user can interact with the GenAI endpoint manually
+	if simConn == nil {
+		go tcpListener(port, s, compConn)
+	} else {
+		go handleChat(*prompt, compConn, simConn, *numChats)
+	}
+
+	<-sigCtx.Done()
+	os.Exit(0)
+}
+
+func initClient(endpoint string, context string) (*connection) {
+	// TODO: create option for a client certificate
+	client := &http.Client{}
+	conn := &connection{client: client, endpoint: endpoint, context: context}
+	return conn
+}
+
+type connection struct {
+	client   *http.Client
+	endpoint string // full base URL for API requests
+	context  string
+	// TODO: create options for routes off the base URL
+}
+
+type GenAIRequest struct {
+	Context         string  `json:"context,omitempty"`
+	Prompt          string  `json:"prompt"`
+}
+
+func handleGenAIRequest(prompt string, clientConn *connection) (string, error) {
+	jsonRequest := GenAIRequest{
+		Context: clientConn.context,
+		Prompt: prompt,
+	}
+	jsonStr, err := json.Marshal(jsonRequest)
+	if err != nil {
+		return "", fmt.Errorf("unable to marshal json request: %v", err)
+	}
+
+	req, err := http.NewRequest("POST", clientConn.endpoint, bytes.NewBuffer(jsonStr))
+	if err != nil {
+		return "unable create http POST request", err
```

Comment:
nit: Use the `return "", fmt.Errorf(...)` pattern here to be consistent with the rest of the function.